app/src/main/java/org/simple/clinic/di/NetworkModule.kt

```diff
@@ -27,10 +28,11 @@ class NetworkModule {
   }
 
   @Provides
-  fun retrofitBuilder(moshi: Moshi, loggedInInterceptor: LoggedInUserHttpInterceptor): Retrofit.Builder {
+  fun retrofitBuilder(moshi: Moshi, loggedInInterceptor: LoggedInUserHttpInterceptor, otpRequestInterceptor: TemporaryValidateOtpRequestInterceptor): Retrofit.Builder {
     val okHttpBuilder = OkHttpClient.Builder()
 
     okHttpBuilder.addInterceptor(loggedInInterceptor)
+    okHttpBuilder.addInterceptor(otpRequestInterceptor)
```

Comment:
Instead of adding this to the global retrofit builder, do you want to keep this logic private to `LoginModule`?

Reply:
Whoops, we can't. The interceptor can only be added to the `Retrofit.Builder` instance and not to the `Retrofit` instance which `LoginModule` depends on.

Reply:
Ah okay. Not very nicely designed, our modules.